lib/pbench/server/api/resources/upload_api.py

```diff
@@ -109,6 +111,14 @@ def put(self, controller):
         md5_full_path = Path(path, f"{filename}.md5")
         bytes_received = 0
 
+        # TODO: Need real user from PUT!
+        name = Dataset.get_name(tar_full_path)
+        self.logger.info(
+            "Uploaded {}:{} for user {}", controller, name, "placeholder",
+        )
+        dataset = Dataset(controller=controller, name=name, md5=md5sum)
+        dataset.add()
```

Comment:
Seems like a simple `dataset.create()` static method is in order, and wrapping it in a `try/except` block?